frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseSlideOutBranchAction.java

```diff
@@ -132,20 +132,16 @@ private void doSlideOut(AnActionEvent anActionEvent, IGitMacheteNonRootBranch br
     new Task.Backgroundable(project, "Deleting branch if required...") {
       @Override
       public void run(ProgressIndicator indicator) {
-        deleteBranchIfRequired(anActionEvent, branchName);
-      }
-
-      @Override
-      public void onFinished() {
-        getGraphTable(anActionEvent).queueRepositoryUpdateAndModelRefresh();
+        deleteBranchIfRequired(anActionEvent, branchName,
+            () -> getGraphTable(anActionEvent).queueRepositoryUpdateAndModelRefresh());
       }
     }.queue();
   }
 
   /**
    * This method must NOT be called on UI thread.
    */
-  private void deleteBranchIfRequired(AnActionEvent anActionEvent, String branchName) {
+  private void deleteBranchIfRequired(AnActionEvent anActionEvent, String branchName, Runnable andThen) {
```

Comment:
Document (javadoc) that `andThen` will NOT be executed on UI thread (IIUC)

Reply:
Actually we can just depend on the topic `GitRepository.GIT_REPO_CHANGE` in case of branch deletion